pkg/controller/node/nodecontroller.go

```diff
@@ -212,7 +213,11 @@ func (nc *NodeController) generateAvailableCIDRs() {
 	// Generate all available CIDRs here, since there will not be manay
 	// available CIDRs. Set will be small, it will use less than 1MB memory
 	cidrIP := nc.clusterCIDR.IP.To4()
-	nc.maxCIDRs = (256-int(cidrIP[1]))*256 - int(cidrIP[2])
+
+	cidrSize, _ := nc.clusterCIDR.Mask.Size()
+
+	// nc.maxCIDRs = int(math.Mod((256-cidr[0])*2^24 + (256-cidr[1])*2^16 + (256-cidr[2])*2^8, 2^(32-cidrSize)) / (2^8))
```

Comment:
The logic itself is short, so instead of going nuclear with #19242 we should fix the logic here if needed.